### ISSUE TITLE
xcatprobe shows false negative result while checking static IP address configuration

### ISSUE THREAD

**Issue by neo954** [xcat2/xcat-core#3825]
This bug is against xCAT latest daily build `xCAT-probe-2.13.7-snap201708300615.noarch`.

The xCAT management node runs RHEL 7.4 on ppc64le.

**The recreation steps**
```
# xcatprobe xcatmn -i enP1p9s0f2
[mn]: Checking all xCAT deamons are running...                                                                    [ OK ]
[mn]: Checking xcatd can receive command request...                                                               [ OK ]
[mn]: Checking 'site' table is configured...                                                                      [ OK ]
[mn]: Checking provision network is configured...                                                                 [ OK ]
[mn]: Checking 'passwd' table is configured...                                                                    [ OK ]
[mn]: Checking important directories(installdir,tftpdir) are configured...                                        [ OK ]
[mn]: Checking SELinux is disabled...                                                                             [ OK ]
[mn]: Checking HTTP service is configured...                                                                      [ OK ]
[mn]: Checking TFTP service is configured...                                                                      [ OK ]
[mn]: Checking DNS service is configured...                                                                       [FAIL]
[mn]: DNS service isn't ready on 10.6.31.1
[mn]: Checking DHCP service is configured...                                                                      [ OK ]
[mn]: Checking NTP service is configured...                                                                       [ OK ]
[mn]: Checking firewall is disabled...                                                                            [ OK ]
[mn]: Checking minimum disk space for xCAT ['/var' needs 1GB;'/install' needs 10GB;'/tmp' needs 1GB]...           [ OK ]
[mn]: Checking xCAT management node IP: <10.6.31.1> is configured to static...                                    [WARN]
[mn]: The value '10.6.31.1' of 'master' in 'site' table isn't a static ip
[mn]: Checking dhcpd.leases file is less than 100M...                                                             [ OK ]
=================================== SUMMARY ====================================
[MN]: Checking on MN...                                                                                           [FAIL]
    Checking DNS service is configured...                                                                         [FAIL]
        DNS service isn't ready on 10.6.31.1
    Checking xCAT management node IP: <10.6.31.1> is configured to static...                                      [WARN]
        The value '10.6.31.1' of 'master' in 'site' table isn't a static ip
```

**What actually happened**
In the output of `xcatprobe` above, it shows the "static IP checking" is failed. And `xcatprobe` shows a `[WARN]` as the result. However, the checking result is incorrect.

The IPv4 address on the server is configured as "static".
```
# ip addr show enP1p9s0f2
6: enP1p9s0f2: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether 0c:c4:7a:f4:1c:56 brd ff:ff:ff:ff:ff:ff
    inet 10.6.31.1/8 brd 10.255.255.255 scope global enP1p9s0f2
       valid_lft forever preferred_lft forever
    inet6 fe80::ec4:7aff:fef4:1c56/64 scope link
       valid_lft forever preferred_lft forever
```
The output below shows what the related configuration file of the network interface looks like.
```
# cat /etc/sysconfig/network-scripts/ifcfg-enP1p9s0f2
# Generated by dracut initrd
NAME="enP1p9s0f2"
DEVICE="enP1p9s0f2"
ONBOOT=yes
NETBOOT=yes
UUID="0bf71d2c-9d31-4754-a18e-1a9b20ddb740"
IPV6INIT=yes
BOOTPROTO=none
IPADDR="10.6.31.1"
NETMASK="255.0.0.0"
GATEWAY="10.0.0.103"
TYPE=Ethernet
```

**Comment by neo954** [comment]
In function `probe_utils->is_static_ip()`, file `xCAT-probe/lib/perl/probe_utils.pm` line 184 and 185, the function checks whether the variable `BOOTPROTO` match value `static`. See source codes below. This checking is incorrect.

```
   175  sub is_static_ip {
   176      my $ip = shift;
   177      $ip = shift if (($ip) && ($ip =~ /probe_utils/));
   178      my $nic = shift;
   179      my $os  = get_os();
   180      my $rst = 0;
   181
   182      if ($os =~ /redhat/) {
   183          my $output1 = `cat /etc/sysconfig/network-scripts/ifcfg-$nic 2>&1 |grep -i IPADDR`;
   184          my $output2 = `cat /etc/sysconfig/network-scripts/ifcfg-$nic 2>&1 |grep -i BOOTPROTO`;
   185          $rst = 1 if (($output1 =~ /$ip/) && ($output2 =~ /static/i));
   186      } elsif ($os =~ /sles/) {
   187          my $output1 = `cat /etc/sysconfig/network/ifcfg-$nic 2>&1 |grep -i IPADDR`;
   188          my $output2 = `cat /etc/sysconfig/network/ifcfg-$nic 2>&1 |grep -i BOOTPROTO`;
   189          $rst = 1 if (($output1 =~ /$ip/) && ($output2 =~ /static/i));
   190      } elsif ($os =~ /ubuntu/) {
   191          my $output = `cat /etc/network/interfaces 2>&1|grep -E "iface\s+$nic"`;
   192          $rst = 1 if ($output =~ /static/i);
   193      }
   194      return $rst;
   195  }
```

**Comment by neo954** [comment]
Please refer the official *NETWORKING GUIDE* for RED HAT ENTERPRISE LINUX. See URLs below. For a network adapter with static configured IP address, the correct setting should be `BOOTPROTO=none`. 

I quote the related parts below.
> *... where the corresponding BOOTPROTO will be set to none and ONBOOT will be set to yes.*

[1] https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/3/html/Reference_Guide/s1-networkscripts-interfaces.html
[2] https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/7/html-single/Networking_Guide/index.html

**Comment by hu-weihua** [comment]
@xuweibj, Could you help to look at above problem? thanks.

**Comment by neo954** [comment]
Verified with `xCAT-probe-2.13.7-snap201709150615.noarch`.

**Comment by neo954** [comment]
Will close this one.

**Comment by dhilst** [comment]
Hi,

Sorry commenting on closed issue, but does NetworkManager is taken in account? I'm using virtual machines for testing xCAT setups and it seems that no `ifcfg-virbr0` is created during the libvirt installation. Still the interface has an static IP as stated by bellow `nmcli` command.

```
[root@vydog-dev ~]# nmcli con show virbr0 | grep -E 'ipv[46].method'
ipv4.method:                            manual
ipv6.method:                            ignore
[root@vydog-dev ~]# 
```

**Comment by whowutwut** [comment]
@xuweibj @neo954 @hu-weihua @bybai please help comment on the above. Thanks!

**Comment by hu-weihua** [comment]
@dhilst, thanks for bringing up this problem.
 
@whowutwut, I opened a task xcat2/xcat2-task-management#154 for above problem . This will be fixed in 2.14.2.

**Comment by robin2008** [comment]
@hu-weihua   If this is an issue,  need to create ticket on xcat2/xcat-core directly, instead of xcat2-task-management.
